lib/OpenQA/Controller/Step.pm

```diff
@@ -391,6 +391,58 @@ sub _timestamp {
     }
 }
 
+sub _json_validation($) {
+
+    my $self = shift;
+    my $json = shift;
+    my %errorh;
+    $errorh{'message'}='';
+    my $djson = eval {decode_json($json)};
+    if (!$djson ) {
+        my $message=$@;
+        $message=~s@at /usr/.*$@@; #do not print perl module reference
+        $errorh{'message'}="syntax error: $message";
+        return \%errorh;
+    }
+
+    if (!exists $djson->{'area'} || !exists $djson->{'area'}[0]) {
+        $errorh{'message'}="no area defined";
+        return \%errorh;
+    }
+    if (!exists $djson->{'tags'} || !exists $djson->{'tags'}[0]) {
+        $errorh{'message'}="no tag defined";
+        return \%errorh;
+    }
+    my $areas=$djson->{'area'};
+    foreach my $area (@$areas) {
+
+        if (!exists $area->{'xpos'} ) {
+            $errorh{'message'}="area without xpos";
+            return \%errorh;
+        }
+        if (!exists $area->{'ypos'} ) {
+            $errorh{'message'}="area without ypos";
+            return \%errorh;
+        }
+        if (!exists $area->{'type'} ) {
+            $errorh{'message'}="area without type";
+            return \%errorh;
+        }
+        if (!exists $area->{'height'} ) {
+            $errorh{'message'}="area without height";
+            return \%errorh;
+        }
+        if (!exists $area->{'width'} ) {
+            $errorh{'message'}="area without width";
+            return \%errorh;
```

Comment:
Idea was to return a richer structure, in the end the message was enough. Not really a problem as it is now, and allows easy extension if needed.